unattended-upgrade

```diff
@@ -610,9 +598,11 @@ def upgrade_in_minimal_steps(cache,            # type: apt.Cache
             return False
         pkg = cache[pkgname]
         if pkg.is_upgradable:
-            pkg.mark_upgrade(from_user=not pkg.is_auto_installed)
+            cache.call_adjusted(apt.package.Package.mark_upgrade, pkg,
```

Comment:
I wonder if it would be cleaner API wise to add a helper to UnattendedUpgradesCache like "cache.mark_upgrade(pkg)" and have "call_adjusted" just as an internal implementation detail in the cache? This would probably make the code in the bits outside the cache easier to read.

Reply:
Done.